java/connector-node/risingwave-sink-mock-flink/risingwave-sink-mock-flink-runtime/src/main/java/com/risingwave/connector/sinkwriter/SinkWriterV2Impl.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2023 RisingWave Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.risingwave.connector.sinkwriter;
+
+import com.risingwave.connector.RowDataImpl;
+import com.risingwave.connector.api.TableSchema;
+import com.risingwave.connector.api.sink.SinkRow;
+import com.risingwave.connector.api.sink.SinkWriterBase;
+import com.risingwave.connector.context.SinkWriterContextV2;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.flink.api.connector.sink2.Committer;
+import org.apache.flink.api.connector.sink2.SinkWriter;
+import org.apache.flink.api.connector.sink2.StatefulSink;
+import org.apache.flink.api.connector.sink2.TwoPhaseCommittingSink;
+import org.apache.flink.connector.base.sink.writer.AsyncSinkWriter;
+import org.apache.flink.table.data.RowData;
+
+/*
+ * The SinkV2-type sink will invoke this class to write RW data to the downstream.
+ */
+public class SinkWriterV2Impl<CommT> extends SinkWriterBase {
+
+    org.apache.flink.api.connector.sink2.Sink<RowData> sink;
+    SinkWriter<RowData> writer;
+
+    Committer<CommT> committer;
+
+    boolean isAsyncSink;
+
+    public SinkWriterV2Impl(
+            TableSchema tableSchema, org.apache.flink.api.connector.sink2.Sink<RowData> sink) {
+        super(tableSchema);
+        try {
+            this.sink = sink;
+            this.writer = sink.createWriter(new SinkWriterContextV2());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        isAsyncSink = writer instanceof AsyncSinkWriter;
+        if (!isAsyncSink && (writer instanceof StatefulSink.StatefulSinkWriter)) {
+            throw new UnsupportedOperationException("Don't support StatefulSink");
+        }
+
+        try {
+            if (this.sink instanceof TwoPhaseCommittingSink) {
```

Comment:
From the sink description, this interface is for exactly-once semantic, which we haven't supported yet. So I think we don't need to adapt to this interface in this PR yet.

https://nightlies.apache.org/flink/flink-docs-master/api/java/org/apache/flink/api/connector/sink2/TwoPhaseCommittingSink.html

Reply:
Yes, However, some sinks use this for committing, and although we haven't implemented two-phase commit, we still need to commit data to downstream sinks.